src/index.js

```diff
@@ -1,42 +1,68 @@
 import { createPlugin } from 'docz-core'
 import * as path from 'path'
+import fs from 'fs-extra'
 import paths from './paths'
+import buildStorybook from './build-storybook'
 
 export const storybook = (opts = { }) => {
   const {
     // these options are intended to mimic the storybook cli options
-    configDir = paths.storybook.configDir
+    configDir = paths.storybook.configDir,
+    manual = false
   } = opts
 
   const storybookConfigPath = path.resolve(configDir, paths.storybook.config)
+  const storybookFiles = []
 
   return createPlugin({
-    setConfig: (config) => {
-      // TODO: since we're only given an entrypoint .storybook/config.js with a configure
-      // call to load stories on the client, we won't know which mdx files to create so
-      // we may have to inject entries dynamically on the client as part of the 'configure'
-      // process. after configure, we add all story kinds to the client-side entries?
+    setConfigPromise: async (config) => {
+      if (!manual) {
+        const storybook = await buildStorybook({ configDir })
+        console.log('storybook', JSON.stringify(storybook, null, 2))
 
-      // TODO: docz will only transpile files from src/* so our temp mdx needs to exist
-      // somewhere in the source tree (not temp), possibly in .docz/storybook
-      // const testMDX = path.join(paths.temp, 'test.mdx')
-      // fs.writeFileSync(testMDX, `import '${paths.storybook.config}'`)
+        await fs.ensureDir(paths.temp)
+        await Promise.all(storybook.map((storyKind) => {
+          const { kind, stories } = storyKind
+
+          const content = `
+---
+name: ${kind} Stories
+---
+
+import { Story } from 'docz-plugin-storybook/dist/react'
+
+# ${kind}
+
+${stories.map(({ name }) => `
+## ${name}
+
+<Story kind='${kind}' name='${name}' />
+
+`)}
+`
```

Comment:
this should probably be separated into a separate template file that's overridable by the user.